iceoryx_platform/win/include/iceoryx_platform/platform_correction.hpp

```diff
@@ -26,11 +26,14 @@
 
 #include <cstring>
 
+#if defined(_MSC_VER)
+
 #ifndef __PRETTY_FUNCTION__
 #define __PRETTY_FUNCTION__ __FUNCSIG__
 #endif
 
 #pragma pointers_to_members(full_generality)
+#endif
```

Comment:
@elfenpiff I managed to get quite far with these compiler switches and there are not that many. If you agree, I would keep it like this and not create a complete new platform for MinGW